src/Agent.Worker/Build/ArtifactCommands.cs

```diff
@@ -0,0 +1,339 @@
+﻿using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Build
+{
+    public sealed class ArtifactCommands : AgentService, ICommandExtension
+    {
+        public Type ExtensionType
+        {
+            get
+            {
+                return typeof(ICommandExtension);
+            }
+        }
+
+        public string CommandArea
+        {
+            get
+            {
+                return "artifact";
+            }
+        }
+
+        public void ProcessCommand(IExecutionContext context, Command command)
+        {
+            if (string.Equals(command.Event, WellKnownArtifactCommand.Associate, StringComparison.OrdinalIgnoreCase))
+            {
+                ProcessArtifactAssociateCommand(context, command.Properties, command.Data);
+            }
+            else if (string.Equals(command.Event, WellKnownArtifactCommand.Upload, StringComparison.OrdinalIgnoreCase))
+            {
+                ProcessArtifactUploadCommand(context, command.Properties, command.Data);
+            }
+            else
+            {
+                throw new Exception(StringUtil.Loc("ArtifactCommandNotFound", command.Event));
+            }
+
+            return;
```

Comment:
Looks like this line was a hold-over from the previous way the method was structured :)